### ISSUE TITLE
An `undefined` value is returned by `OverlayRef.detach()`

### ISSUE THREAD

**Issue by dherges** [angular/material2#7408]
#### Bug, feature request, or proposal:

Bug

#### What is the expected behavior?

```ts
export declare class OverlayRef implements PortalHost {
    /**
     * Detaches an overlay from a portal.
     * @returns Resolves when the overlay has been detached.
     */
    detach(): Promise<any>;
```

#### What is the current behavior?

An `undefined` value is returned by `OverlayRef.detach()`

#### What are the steps to reproduce?

![2017-09-29 10_04_41-developer tools - http___localhost_4200_](https://user-images.githubusercontent.com/901354/31006740-4e7662d8-a4fe-11e7-9b10-3dc51055b123.png)

StackBlitz: https://angular-material2-issue-qqyacc.stackblitz.io/

StackBlitz export: [angular-material2-issue.zip](https://github.com/angular/material2/files/1343620/angular-material2-issue.zip)

The plunker starter btw errors with no modifications:

![2017-09-29 10_17_11-developer tools - http___plnkr co_edit_o077b6ueiiigkc0s06dd_p preview](https://user-images.githubusercontent.com/901354/31007200-007bf69a-a500-11e7-87d3-f164d0450a99.png)

#### What is the use-case or motivation for changing an existing behavior?

Is it a bug?

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

`@angular/cdk: 2.0.0-beta.11`
